### PR TITLE
Fix preprocessing in MO to use friendly names as well as tensor names

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Preprocessing in ngraph relies on tensor names, but MO users don't expect that, so for backward compatibility we need to look into friendly names as well*

### Tickets:
 - 73794

### CODE REVIEW

File: tools/mo/openvino/tools/mo/back/preprocessing.py [hunk 1]
```diff
@@ -80,13 +82,26 @@ def check_keys_valid(ov_function: Model, keys: list, search_outputs: bool):
...
+
 
         if not node_found:
             if not search_outputs:
                 raise Error('Input with name {} wasn\'t found! {}'.format(name, refer_to_faq_msg(83)))
             else:
                 raise Error('Input/Output with name {} wasn\'t found! {}'.format(name, refer_to_faq_msg(83)))
 
+    for name, new_name in rename_dict.items():
+        if name in dict_to_validate:
```
Comment (nosovmik) [inline tools/mo/openvino/tools/mo/back/preprocessing.py:101]: i guess `name` is always in 'dict_to_validate` here, so we can add 'assert' instead

File: tools/mo/openvino/tools/mo/back/preprocessing.py [hunk 2]
```diff
@@ -328,7 +343,7 @@ def apply_preprocessing(ov_function: Model, argv: argparse.Namespace):
 
     mean_scale_values = update_mean_scale_to_dict(input_nodes=ov_function.inputs,
                                                   mean_scale_val=mean_scale_values,
-                                                  scale=argv.scale)
+                                                  scale=argv.scale if "scale" in argv else None)
```
Comment (nosovmik) [inline tools/mo/openvino/tools/mo/back/preprocessing.py:346]: is this needed? Previously 'argv' always had 'scale' attribute and there was no need for such check. Is it not correct  anymore?

**Review by nosovmik — APPROVED** [review]
Looks good to me, only few minor comments

File: tools/mo/openvino/tools/mo/back/preprocessing.py (hunk 2 above)
Comment (mvafin) [reply tools/mo/openvino/tools/mo/back/preprocessing.py:346]: you are right, will delete

**Review by artyomtugaryov — APPROVED** [review]

**Comment by dkurt** [comment]
Hi! Can this PR be merged?

**Comment by rkazants** [comment]
> Hi! Can this PR be merged?

Let me review this PR first please

File: tools/mo/openvino/tools/mo/back/preprocessing.py [hunk 3]
```diff
@@ -70,7 +70,9 @@ def check_keys_valid(ov_function: Model, keys: list, search_outputs: bool):
     if search_outputs:
         nodes += ov_function.outputs
 
-    for name in keys:
+    rename_dict = {}
+
+    for name in dict_to_validate.keys():
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/back/preprocessing.py:75]: I propose to do this within two steps:
1) rename keys in dictionary if needed
2) perform this check

Comment (rkazants) [reply tools/mo/openvino/tools/mo/back/preprocessing.py:75]: Also, please add comments for each pass about why it is needed

File: tools/mo/openvino/tools/mo/back/offline_transformations.py [hunk 4]
```diff
@@ -104,7 +59,28 @@ def read_model(path_to_xml):
 
     func = read_model(input_model + "_tmp.xml")
 
-    add_layouts(func, argv)  # TODO: replace with preprocessing
+    reverse_input_channels = False
+    if 'reverse_input_channels' in argv:
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/back/offline_transformations.py:63]: please add TODO that we reset this options temporarily. Now we still use legacy preprocessing

**Review by rkazants — APPROVED** [review]

**Pull request merged**